src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordDeconstruct.cs

```diff
@@ -15,135 +15,83 @@
-
-        public override ImmutableArray<CustomModifier> RefCustomModifiers => ImmutableArray<CustomModifier>.Empty;
-
-        public override Symbol? AssociatedSymbol => null;
-
-        internal override CallingConvention CallingConvention => CallingConvention.HasThis;
-
-        internal override bool GenerateDebugInfo => false;
-
-        public override Symbol ContainingSymbol => ContainingType;
-
-        public override ImmutableArray<Location> Locations => ContainingType.Locations;
-
-        public override Accessibility DeclaredAccessibility => Accessibility.Public;
-
-        public override bool IsStatic => false;
-
-        public override bool IsVirtual => false;
-
-        public override bool IsOverride => false;
-
-        public override bool IsAbstract => false;
-
-        public override bool IsSealed => false;
-
-        public override bool IsExtern => false;
+        protected override DeclarationModifiers MakeDeclarationModifiers(DeclarationModifiers allowedModifiers, DiagnosticBag diagnostics)
+        {
+            const DeclarationModifiers result = DeclarationModifiers.Public;
+            Debug.Assert((result & ~allowedModifiers) == 0);
+            return result;
+        }
 
-        internal override LexicalSortKey GetLexicalSortKey() => LexicalSortKey.GetSynthesizedMemberKey(_memberOffset);
+        protected override (TypeWithAnnotations ReturnType, ImmutableArray<ParameterSymbol> Parameters, bool IsVararg, ImmutableArray<TypeParameterConstraintClause> DeclaredConstraintsForOverrideOrImplementation) MakeParametersAndBindReturnType(DiagnosticBag diagnostics)
+        {
+            var compilation = DeclaringCompilation;
+            var location = ReturnTypeLocation;
+            return (ReturnType: TypeWithAnnotations.Create(Binder.GetSpecialType(compilation, SpecialType.System_Void, location, diagnostics)),
+                    Parameters: _ctor.Parameters.SelectAsArray<ParameterSymbol, ImmutableArray<Location>, ParameterSymbol>(
+                                        (param, locations) =>
+                                            new SourceSimpleParameterSymbol(owner: this,
+                                                param.TypeWithAnnotations,
+                                                param.Ordinal,
+                                                RefKind.Out,
+                                                param.Name,
+                                                isDiscard: false,
+                                                locations),
+                                        arg: Locations),
+                    IsVararg: false,
+                    DeclaredConstraintsForOverrideOrImplementation: ImmutableArray<TypeParameterConstraintClause>.Empty);
+        }
 
-        internal override bool SynthesizesLoweredBoundBody => true;
+        protected override int GetParameterCountFromSyntax() => _ctor.ParameterCount;
```

Comment:
>GetParameterCountFromSyntax [](start = 31, length = 27)

This is strange. I don't think it makes sense to ask how many parameters the `Deconstruct` method has in syntax, given that it has no syntax.
I think we should override `ParameterCount` and throw in this method (unreachable)

Reply:
Why "won't fix"?

---
In reply to: [456676237](https://github.com/dotnet/roslyn/pull/46053#discussion_r456676237) [](ancestors = 456676237)

Reply:
> Why "won't fix"?

I find nothing strange in this name. It was not introduced in this PR. This is just an abstraction to give out the number of parameters without creating parameters themselves. Completely makes sense to me for any method. I can see that you might want this method to have a different name, but changing it is not in scope for this PR. Overriding would be the wrong thing to do, because we want it to do exactly what it does today, even for this method.  

---
In reply to: [456689715](https://github.com/dotnet/roslyn/pull/46053#discussion_r456689715) [](ancestors = 456689715,456676237)